main.py

```diff
@@ -0,0 +1,31 @@
+import player, domino_tile, game_data as gd, game_logic as gl
+
+def number_of_players_validation():
+  while(True):
+    number_of_players = input()
+    if len(number_of_players) == 1 and number_of_players.isdigit() and (int(number_of_players) > 1 and int(number_of_players) < 5):
```

Comment:
You can also create a function to set this condition more understandable sometimes like:
```
if (ConditionA and ConditionB)
```

This is better 'cause you will set a good name to each condition, so if the condition is `(int(number_of_players) > 1 and int(number_of_players) < 5)` you can call it as `areEnoughPlayers`, so here I know that here you are looking for the correct players' number.

You can do it everywhere you think to have a lot of conditions

Reply:
Also here I can set 3 players, is it allowed by rules?